.github/workflows/checkupstream.yml

```diff
@@ -0,0 +1,32 @@
+name: Check new upstream commits
```

Comment:
This doesn't seem to be relevant to this PR?

Reply:
Sorry, my bad. I had had to create another branch for this PR. I thought it is abandoned and pushed mechanism informing me about new commits.
Anyway, what do you think about the changes? Worth merging? Without last commit of course :)